cmd/snap/cmd_run_test.go

```diff
@@ -72,20 +73,26 @@ func (s *SnapSuite) TestSnapRunSnapExecEnv(c *check.C) {
 	usr, err := user.Current()
 	c.Assert(err, check.IsNil)
 
-	env := snaprun.SnapExecEnv(info)
-	sort.Strings(env)
-	c.Check(env, check.DeepEquals, []string{
-		fmt.Sprintf("SNAP=%s/snapname/42", dirs.SnapMountDir),
-		fmt.Sprintf("SNAP_ARCH=%s", arch.UbuntuArchitecture()),
-		"SNAP_COMMON=/var/snap/snapname/common",
-		"SNAP_DATA=/var/snap/snapname/42",
-		"SNAP_LIBRARY_PATH=/var/lib/snapd/lib/gl:",
-		"SNAP_NAME=snapname",
-		"SNAP_REVISION=42",
-		fmt.Sprintf("SNAP_USER_COMMON=%s/snap/snapname/common", usr.HomeDir),
-		fmt.Sprintf("SNAP_USER_DATA=%s/snap/snapname/42", usr.HomeDir),
-		"SNAP_VERSION=1.0",
-	})
+	for _, withHomeEnv := range []bool{true, false} {
+		if !withHomeEnv {
+			os.Setenv("HOME", "")
```

Comment:
Doesn't this set the environment for **all** tests? Should it be undone at the end? Perhaps I misunderstand that.